iroh-relay/src/main.rs

```diff
@@ -181,6 +187,27 @@ enum AccessConfig {
     Allowlist(Vec<NodeId>),
     /// Allows everyone, except these nodes.
     Denylist(Vec<NodeId>),
+    /// Performs a HTTP POST request to determine access for each node that connects to the relay.
+    ///
+    /// The request will have a header `X-Iroh-Node-Id` set to the hex-encoded node id attempting
+    /// to connect to the relay.
+    ///
+    /// To grant access, the HTTP endpoint must return a `200` response with `true` as the response text.
+    /// In all other cases, the node will be denied access.
+    Http(HttpAccessConfig),
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+struct HttpAccessConfig {
```

Comment:
I'm always nervous whether serde manages to de-serialise from the config file format correctly.  I've seen cases where serde expressed things that could not be expressed in the target serialisation.  Mind adding a de-serialisation test for this from toml?  (We don't need serialise support on these anymore, they're historic.)

There are a few similar tests already, so should be easy enough to copy.